akd/src/append_only_zks.rs

```diff
@@ -41,6 +47,26 @@ async fn tic_toc<T>(f: impl core::future::Future<Output = T>) -> (T, Option<f64>
     (f.await, None)
 }
 
+fn get_parallel_levels() -> Option<u8> {
+    #[cfg(not(feature = "parallel_insert"))]
+    return None;
+
+    #[cfg(feature = "parallel_insert")]
+    {
+        // Based on profiling results, the best performance is achieved when the
+        // number of spawned tasks is equal to the number of available cores.
+        // We therefore get the number of available threads and calculate the
+        // number of levels that should be executed in parallel to give the
+        // number of tasks closest to the number of threads. While there might
+        // be other tasks that are running on the threads, this is a reasonable
+        // approximation that should yield good performance in most cases.
+        let available_parallelism = std::thread::available_parallelism()
+            .map_or(DEFAULT_AVAILABLE_PARALLELISM, |v| v.into());
+        let levels = (available_parallelism as f32).log2().ceil() as u8;
```

Comment:
nit: a not about why a log_2 here would be helpful (something like because it's a binary tree and at each level we spawn 2 threads / parent node